__tests__/login/oidc/refresh/TokenRefresher.spec.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import "reflect-metadata";
+import TokenRefresher from "../../../../src/login/oidc/refresh/TokenRefresher";
+import { FetcherMock } from "../../../../src/util/__mocks__/Fetcher";
+import {
+  IssuerConfigFetcherMock,
+  IssuerConfigFetcherFetchConfigResponse
+} from "../../../../src/login/oidc/__mocks__/IssuerConfigFetcher";
+import { StorageUtilityMock } from "../../../../src/localStorage/__mocks__/StorageUtility";
+import {
+  DpopHeaderCreatorMock,
+  DpopHeaderCreatorResponse
+} from "../../../../src/dpop/__mocks__/DpopHeaderCreator";
+
+describe("TokenRefresher", () => {
+  const defaultMocks = {
+    fetcher: FetcherMock,
+    issueConfigFetcher: IssuerConfigFetcherMock,
+    storageUtility: StorageUtilityMock,
+    dpopHeaderCreator: DpopHeaderCreatorMock
+  };
+  function getTokenRefresher(
+    mocks: Partial<typeof defaultMocks> = defaultMocks
+  ): TokenRefresher {
+    return new TokenRefresher(
+      mocks.fetcher ?? defaultMocks.fetcher,
+      mocks.issueConfigFetcher ?? defaultMocks.issueConfigFetcher,
+      mocks.storageUtility ?? defaultMocks.storageUtility,
+      mocks.dpopHeaderCreator ?? defaultMocks.dpopHeaderCreator
+    );
+  }
+
```

Comment:
As mentioned in the class itself: it would be nice to also have a test in which the fetcher returns invalid JSON, or the request fails, and verifies that the resulting error is meaningful to the developer and can be gracefully dealt with.